builtin/mainmenu/dlg_settings_advanced.lua

```diff
@@ -501,15 +501,16 @@ local function get_current_np_group(setting)
 	if value == nil then
 		t = setting.values
 	else
-		table.insert(t, value.offset)
-		table.insert(t, value.scale)
-		table.insert(t, value.spread.x)
-		table.insert(t, value.spread.y)
-		table.insert(t, value.spread.z)
-		table.insert(t, value.seed)
-		table.insert(t, value.octaves)
-		table.insert(t, value.persistence)
-		table.insert(t, value.lacunarity)
+		local p = "%g"
+		table.insert(t, p:format(value.offset))
+		table.insert(t, p:format(value.scale))
+		table.insert(t, p:format(value.spread.x))
+		table.insert(t, p:format(value.spread.y))
+		table.insert(t, p:format(value.spread.z))
+		table.insert(t, p:format(value.seed))
+		table.insert(t, p:format(value.octaves))
+		table.insert(t, p:format(value.persistence))
+		table.insert(t, p:format(value.lacunarity))
 		table.insert(t, value.flags)
```

Comment:
btw this is complicated for no reason, it's better written as
```lua
    t = {
        p:format(value.offset),
        p:format(value.scale),
        -- ...
        p:format(value.lacunarity),
        value.flags
    }
```

Reply:
Right, and it could have been this way since the beginning. Rewritten.